akka-stream/src/main/scala/akka/stream/scaladsl/One2OneBidiFlow.scala

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.scaladsl
+
+import akka.stream._
+
+import scala.util.control.NoStackTrace
+
+/**
+ * A generic BidiFlow which verifies that another flow produces exactly one output element per
```

Comment:
Unfortunately I don't think we can reliably distinguish

```
output: 1
input: 1
```

from

```
input: 1
output: 1
```

at least not with the current bypass-based design.
We could do it with an `AtomicInteger` or something similar, but then we'd have to solve the several-materializations problem.

Do you have a good alternative design in mind?